git-keeper.py

```diff
@@ -24,63 +24,82 @@
 tar = sh.tar.bake('-cf')
 workdir = 'workdir'
 
+
 def upload2s3(s3_client, repo_tar, git_keeper_bucket, date, object_name):
     try:
-        response = s3_client.upload_file(repo_tar, git_keeper_bucket, date + '/' + object_name)
+        s3_client.upload_file(
+            repo_tar, git_keeper_bucket, date + '/' + object_name)
     except ClientError as e:
         logging.error(e)
         sys.exit(1)
 
+
 def cleanwrkdir(workdir):
     shutil.rmtree(workdir, ignore_errors=True)
     os.makedirs(workdir, exist_ok=True)
 
+
 def clone_repo(repo_url, repo_dir):
     mirror(repo_url, repo_dir)
 
+
+def check_aws_creds(aws_access_key_id, aws_secret_access_key):
+    try:
+        boto3.client('s3',
+                     aws_access_key_id=aws_access_key_id,
+                     aws_secret_access_key=aws_secret_access_key)
+    except botocore.exceptions.NoCredentialsError:
+        raise Exception('No AWS credentials found.')
+    except botocore.exceptions.ClientError:
+        raise Exception('Invalid AWS credentials.')
+
+
+def git_clone_upload(s3_client, gpg, recipients, repo, s3_bucket, date):
+    repo_url = repo + '.git'
+    repo_dir = workdir + '/' + os.path.basename(repo_url)
+    repo_tar = repo_dir + '.tar'
+    cleanwrkdir(workdir)
+    clone_repo(repo_url, repo_dir)
+    tar(repo_tar, repo_dir)
+    repo_gpg = repo_tar + '.gpg'
+    with open(repo_tar, 'rb') as f:
+        gpg.encrypt_file(
+            f, recipients=recipients,
+            output=repo_gpg,
+            armor=False,
+            always_trust=True)
+    upload2s3(s3_client, repo_gpg, s3_bucket,
+              date, os.path.basename(
+                  repo_url) + '.tar.gpg')
+    cleanwrkdir(workdir)
+
+
 def main():
     parser = argparse.ArgumentParser(
         description='Configuration TOML and GPG keys locations.')
     parser.add_argument('--config', type=str,
-        help='Path of configuration TOML file')
+                        help='Path of configuration TOML file')
     parser.add_argument('--gpgs', type=str,
-        help='Path of GPG keys file')
+                        help='Path of GPG keys file')
     args = parser.parse_args()
     cnf = toml.load(open(args.config))
-    DATE = datetime.now().strftime('%Y-%m-%d')
-    try:
-        s3_client = boto3.client('s3',
-            aws_access_key_id = cnf["s3"]["aws_access_key_id"],
-            aws_secret_access_key = cnf["s3"]["aws_secret_access_key"])
-    except botocore.exceptions.NoCredentialsError:
-        raise Exception('No AWS credentials found.')
-    except botocore.exceptions.ClientError:
-        raise Exception('Invalid AWS credentials.')
+    aws_access_key_id = cnf["s3"]["aws_access_key_id"]
+    aws_secret_access_key = cnf["s3"]["aws_secret_access_key"]
+    s3_bucket = cnf["s3"]["bucket"]
+    date = datetime.now().strftime('%Y-%m-%d')
```

Comment:
can you add empty new lines so it's not all grouped together, but split into logical groups?

```
# get s3 client
...

# read configuration
...

# set the date
....
```